framework/src/wifi_manager/wifi_profile.c

```diff
@@ -0,0 +1,238 @@
+
+	if (config->ap_auth_type == WIFI_MANAGER_AUTH_OPEN) {
+		return strlen(buf) + 1;
+	}
+
+	ENCODE_STRING(buf, buf_size, config->passphrase, pos);
+	ENCODE_INTEGER(buf, buf_size, config->passphrase_length, pos);
+
+	int crypto_type = (int)config->ap_crypto_type;
+	ENCODE_INTEGER(buf, buf_size, crypto_type, pos);
+
+	return strlen(buf) + 1;
+}
+
+void _wifi_profile_deserialize(wifi_manager_ap_config_s *config, char *buf)
+{
+	int pos = 0;
+	DECODE_STRING(buf, config->ssid, pos);
+	DECODE_INTEGER(buf, config->ssid_length, pos);
+	int auth_type = 0;
+	DECODE_INTEGER(buf, auth_type, pos);
+	config->ap_auth_type = (wifi_manager_ap_auth_type_e)auth_type;
+	if (config->ap_auth_type == WIFI_MANAGER_AUTH_OPEN) {
+		return;
+	}
+	DECODE_STRING(buf, config->passphrase, pos);
+	DECODE_INTEGER(buf, config->passphrase_length, pos);
+	int crypto_type = 0;
+	DECODE_INTEGER(buf, crypto_type, pos);
+	config->ap_crypto_type = (wifi_manager_ap_crypto_type_e)crypto_type;
+}
+
+#ifdef CONFIG_WIFI_PROFILE_SECURESTORAGE
+int _wifi_profile_store_file(char *buf, unsigned int buf_size)
+{
+	FILE *fp = fopen(WIFI_PROFILE_PATH WIFI_PROFILE_FILENAME, "w+");
+	if (!fp) {
+		ndbg("file open error(%d)\n", errno);
+		return -1;
+	}
+
+	int ret = fwrite(buf, 1, buf_size, fp);
+	if (ret < 0) {
+		ndbg("file write error(%d)\n", errno);
+		fclose(fp);
+		return -1;
+	}
+	fclose(fp);
+	return 0;
+}
+
+
+int _wifi_profile_read_file(char *buf, unsigned int buf_size)
+{
+	FILE *fp = fopen(WIFI_PROFILE_PATH WIFI_PROFILE_FILENAME, "r");
+	if (!fp) {
+		ndbg("file open error(%d)\n", errno);
+		return -1;
+	}
+
+	int ret = fread(buf, 1, buf_size, fp);
+	if (ret < 0) {
+		ndbg("fread fail\n");
+		fclose(fp);
+		return -1;
+	}
+	fclose(fp);
+	return 0;
+}
+#endif
+
+/*
+ * Public Functions
+ */
+
+wifi_utils_result_e wifi_profile_init(void)
+{
+#ifdef CONFIG_WIFI_PROFILE_SECURESTORAGE
+#ifdef WIFI_PROFILE_USE_ETC
+	// Temporary code. /etc should be created as default
+	DIR *dir = opendir(WIFI_PROFILE_PATH);
+	if (!dir) {
+		ndbg("error reason (%d)\n", errno);
+		if (errno == ENOENT || errno == ENOTDIR) {
+			// create file
+			ret = mkdir(WIFI_PROFILE_PATH, 0777);
+			if (ret < 0) {
+				return WIFI_UTILS_FILE_ERROR;
+			}
+		} else {
+			return WIFI_UTILS_FILE_ERROR;
+		}
+	}
+#endif
+#endif
+	return WIFI_UTILS_SUCCESS;
+}
+
+wifi_utils_result_e wifi_profile_reset(void)
+{
+	int ret = -1;
+#ifdef CONFIG_WIFI_PROFILE_SECURESTORAGE
+	char buf = '\0';
+	ret = see_write_secure_storage((unsigned char *)&buf, (unsigned int)1, WIFI_PROFILE_SS_INDEX);
+	if (ret != SEE_OK) {
+		ndbg("Write SS fail(%d)\n", ret);
+		return WIFI_UTILS_FILE_ERROR;
+	}
+#else
+	ret = unlink(WIFI_PROFILE_PATH WIFI_PROFILE_FILENAME);
+	if (ret < 0) {
+		ndbg("Delete Wi-Fi profile fail(%d)\n", errno);
+		return WIFI_UTILS_FILE_ERROR;
+	}
+#endif
+	return WIFI_UTILS_SUCCESS;
+}
+
+wifi_utils_result_e wifi_profile_write(wifi_manager_ap_config_s *config)
+{
+	char buf[256];
+	int ret = 0, len = 0;
```

Comment:
Please define a new variable in different line. That is a coding rule recommandation.

Reply:
Ok I'll fix